src/sample.js

```diff
@@ -365,5 +365,78 @@ export default {
   'book': $B.title,
   'due': $L.return
 }`
+    },
+    Posts: {
+        json: {
+            posts : {
+                byId : {
+                    "post1" : {
+                        id : "post1",
+                        author : "user1",
+                        body : "... user 1 ...",
+                        comments : ["comment1", "comment2"]
+                    },
+                    "post2" : {
+                        id : "post2",
+                        author : "user2",
+                        body : "... user 2 ...",
+                        comments : ["comment3", "comment4", "comment5"]
+                    }
+                },
+                allIds : ["post1", "post2"]
+            },
+            comments : {
+                byId : {
+                    "comment1" : {
+                        id : "comment1",
+                        author : "user2",
+                        comment : "... comment 1 ...",
+                    },
+                    "comment2" : {
+                        id : "comment2",
+                        author : "user3",
+                        comment : "... comment 2 ...",
+                    },
+                    "comment3" : {
+                        id : "comment3",
+                        author : "user3",
+                        comment : "... comment 3 ...",
+                    },
+                    "comment4" : {
+                        id : "comment4",
+                        author : "user1",
+                        comment : "... comment 4 ...",
+                    },
+                    "comment5" : {
+                        id : "comment5",
+                        author : "user3",
+                        comment : "... comment 5 ...",
+                    },
+                },
+                allIds : ["comment1", "comment2", "comment3", "commment4", "comment5"]
+            },
+            users : {
+                byId : {
+                    "user1" : {
+                        username : "user1",
+                        name : "User 1",
+                    },
+                    "user2" : {
+                        username : "user2",
+                        name : "User 2",
+                    },
+                    "user3" : {
+                        username : "user3",
+                        name : "User 3",
+                    }
+                },
+                allIds : ["user1", "user2", "user3"]
+            }
+        },
+        jsonata: `(
+    $comment := function($cid) {( comments.byId.$lookup($cid) )};
+    $post := function($pid) {( posts.byId.$lookup($pid) )};
+    posts.allIds{$: $post($).comments.$comment($)}
```

Comment:
This is good, but I would have used the following expression for the same effect:
```
posts.allIds@$P{
    $P: byId.$lookup($P).comments.$lookup($$.comments.byId, $)
}
```

alternatively, if you can assume that the list of post IDs in `posts.allIds` is the same as entries in `posts.byId` then you could simplify it:

```
posts.byId.*{
    id: comments.$lookup($$.comments.byId, $)
}
```